### ISSUE TITLE
Player with signedit.use is able to edit all signs regardless of his permissions to the WorldGuard region

### ISSUE THREAD

**Issue by oskarkk** [Deltik/SignEdit#26]
I'm using the newest version of the plugin with the newest Paper and WorldGuard. Player with signedit.use is able to edit a sign even in a region in which he can't place or break any blocks. I've tried with both `edit-validation: "Extra"` and `edit-validation: "Standard"`. I've restarted the server after editing the config.

**Comment by Deltik** [comment]
@oskarkk, I am not able to reproduce your issue.

Test software:
* **Paper** [git-Paper-381](https://papermc.io/api/v2/projects/paper/versions/1.17.1/builds/381/downloads/paper-1.17.1-381.jar) (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: db0bff9)
* **LuckPerms** [v5.3.75](https://ci.lucko.me/job/LuckPerms/1374/artifact/bukkit/loader/build/libs/LuckPerms-Bukkit-5.3.75.jar)
* **SignEdit** [v1.13.3](https://github.com/Deltik/SignEdit/releases/download/v1.13.3/SignEdit-1.13.3.jar)
* **WorldEdit** [v7.2.7+9f3e795](https://dev.bukkit.org/projects/worldedit/files/3502099/download)
* **WorldGuard** [v7.0.6+1cff55f](https://dev.bukkit.org/projects/worldguard/files/3461546/download)

Configuration:
* In `plugins/WorldGuard/config.yml`, set:
  ```yaml
  build-permission-nodes:
      enable: true
      deny-message: '&cBug test: Deltik/SignEdit#26'
  ```
* Set permissions for my Minecraft user:
  ```
  lp user deltik permission set SignEdit.use true
  [17:16:29 INFO]: [LP] Set SignEdit.use to true for deltik in context global.
  ```
  Resultant permissions:
  ```
  lp user Deltik permission info
  [17:24:07 INFO]: [LP] deltik's Permissions:  (page 1 of 1 - 1 entries)
  [17:24:07 INFO]: > SignEdit.use
  ```

![image](https://user-images.githubusercontent.com/1364268/141702661-168732a3-66cc-4078-8ee9-fca069ac42a9.png)

```
lp user deltik permission set SignEdit.use true
[17:16:29 INFO]: [LP] Set SignEdit.use to true for deltik in context global.
```

**Comment by Deltik** [comment]
@oskarkk, I am not able to reproduce your issue.

Test software:
* **Paper** [git-Paper-381](https://papermc.io/api/v2/projects/paper/versions/1.17.1/builds/381/downloads/paper-1.17.1-381.jar) (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: db0bff9)
* **LuckPerms** [v5.3.75](https://ci.lucko.me/job/LuckPerms/1374/artifact/bukkit/loader/build/libs/LuckPerms-Bukkit-5.3.75.jar)
* **SignEdit** [v1.13.3](https://github.com/Deltik/SignEdit/releases/download/v1.13.3/SignEdit-1.13.3.jar)
* **WorldEdit** [v7.2.7+9f3e795](https://dev.bukkit.org/projects/worldedit/files/3502099/download)
* **WorldGuard** [v7.0.6+1cff55f](https://dev.bukkit.org/projects/worldguard/files/3461546/download)

Configuration:
* In `plugins/WorldGuard/config.yml`, set:
  ```yaml
  build-permission-nodes:
      enable: true
      deny-message: '&cBug test: Deltik/SignEdit#26'
  ```
* Set permissions for my Minecraft user:
  ```
  lp user deltik permission set SignEdit.use true
  [17:16:29 INFO]: [LP] Set SignEdit.use to true for deltik in context global.
  ```
  Resultant permissions:
  ```
  lp user Deltik permission info
  [17:24:07 INFO]: [LP] deltik's Permissions:  (page 1 of 1 - 1 entries)
  [17:24:07 INFO]: > SignEdit.use
  ```

When I ran `/se set 1-4 Deltik/SignEdit#26` on a sign, I got the WorldGuard error message `§cBug test: Deltik/SignEdit#26`, and the sign contents were not changed:

![NOREPRO Deltik/SignEdit#26](https://user-images.githubusercontent.com/1364268/141702661-168732a3-66cc-4078-8ee9-fca069ac42a9.png)

It is likely that you have a permissions configuration mistake on your server.  For this reason, I'll close this issue, but if you believe that there is a bug, please provide more information on how to reproduce it, and I'll reopen this issue.

**Comment by oskarkk** [comment]
Okay, thanks, I'll test it some more.

**Comment by oskarkk** [comment]
Okay, with: 

```
build-permission-nodes:
    enable: true
```

... player can't change the sign, but I don't use build-permission-nodes, I just use regions. And the player can't do anything in the region I am testing this in, but he can change the signs via SignEdit. So I guess SignEdit just isn't compatible with WorldGuard?

**Comment by Deltik** [comment]
@oskarkk: I am still unable to reproduce the issue after setting `build-permission-nodes.enable=false` and trying to edit a sign in a plain WorldGuard region:

![NOREPRO Deltik/SignEdit#26](https://user-images.githubusercontent.com/1364268/141715141-5797d4b1-80fb-4f84-b311-314841dbafe3.png)

The error message is different, but the sign edit is still forbidden.  This looks like neither a SignEdit for Bukkit bug nor a WorldGuard bug.

**Comment by oskarkk** [comment]
I found that it's caused by `interact` flag set to allow. With interact flag, player can't place or destroy signs, but can edit the sign via `/signedit`. I thought that when player can't break/place signs, he shouldn't be able to edit them, especially with `edit-validation: "Extra"` (beacuse they "emit BlockBreakEvent, BlockPlaceEvent and SignChangeEvent (...)").

And for reference I've just checked the behaviour of the EssentialsX `/editsign` command - it doesn't check for any region permissions, so even right now SignEdit wins.

In a clean region without flags the plugin behaves just like on your screenshots. Sorry for not providing the details of my region from the start.

**Comment by Deltik** [comment]
@oskarkk: Indeed, when I run `rg flag -w world bug interact allow`, I can edit signs in the WorldGuard region.

But when I set `compatibility.edit-validation: Extra` in `plugins/SignEdit/config.yml`, the sign edit is once again forbidden, this time with a third error message text (`Hey! Sorry, but you can't break that block here.`):

![NOREPRO Deltik/SignEdit#26](https://user-images.githubusercontent.com/1364268/141722515-19c8c348-7863-4248-82bf-0565d892ba6e.png)

So as far as I can tell, both SignEdit for Bukkit and WorldGuard are doing exactly what's documented.

**Comment by Deltik** [comment]
Oh, you know what, @oskarkk?  I'm not getting the sign edit validation with `/sign ui`.  This whole time, I had been testing with `/sign set` or `/sign clear`.  So this might actually be a bug with SignEdit for Bukkit.  I'm reopening this issue.

**Comment by oskarkk** [comment]
Sorry again for not specifying the exact command I used - and it was `/signedit ui` indeed, all the time. I haven't thought that permission checking for the signs could be different between these subcommands.

**Comment by Deltik** [comment]
`/sign ui` ignoring the `compatibility.edit-validation: Extra` configuration option was a regression caused by https://github.com/Deltik/SignEdit/commit/3b1d590e3add7c2a3c68045c5d59719416b1696d and landed in [v1.12.2](https://github.com/Deltik/SignEdit/releases/tag/v1.12.2).

The bug that was fixed was "The `SignChangeEvent` emitted natively by the native sign editor GUI is no longer sent twice.", but applying this fix neglected the break-replace-edit validator for `/sign ui`.

**Comment by Deltik** [comment]
Yeah, the permission checking had to be different because for `/sign ui`, a `SignChangeEvent` is given to the plugin by Bukkit, but for the other commands like `/sign set`, `/sign cut`, and `/sign undo`, the plugin has to make its own `SignChangeEvent`.

**Comment by Deltik** [comment]
@oskarkk: Thank you for your patience!

The reported bug has now been fixed in [SignEdit for Bukkit v1.13.4](https://github.com/Deltik/SignEdit/releases/tag/v1.13.4), which you can directly download [here](https://github.com/Deltik/SignEdit/releases/download/v1.13.4/SignEdit-1.13.4.jar).

**Comment by oskarkk** [comment]
Great, thank you for a fast response 🙂
